src/server.c

```diff
@@ -572,27 +572,33 @@ static void monitor_cb(uv_prepare_t *monitor)
 {
 	struct dqlite_node *d = monitor->data;
 	int state = raft_state(&d->raft);
-	/*
 	queue *head;
 	struct conn *conn;
-	*/
+	struct gateway *g;
 
 	if (state == RAFT_UNAVAILABLE) {
 		return;
 	}
 
-	/* TODO: we should shutdown clients that are performing SQL requests,
-	 * but not the ones which are doing management-requests, such as
-	 * transfer leadership.  */
-	/*
 	if (d->raft_state == RAFT_LEADER && state != RAFT_LEADER) {
+		tracef("leadership lost, close leader connections");
 		QUEUE__FOREACH(head, &d->conns)
 		{
 			conn = QUEUE__DATA(head, struct conn, queue);
-			conn__stop(conn);
+			g = &conn->gateway;
+			if (g == NULL || g->leader == NULL) {
+				continue;
+			}
+
+			if (g->req != NULL && g->stmt == NULL) {
```

Comment:
Are we really sure that we don't want to have this check read:

```
if (g->req != NULL && g->req->type == REQUEST_TRANSFER) {
```

or add a sort of "whitelist" of request types that we don't want to kill immediately.

I didn't really think too much about the implications of not shutting down other management requests, but perhaps just going through all of them and considering the implications of this change would be good (if you didn't already).

Reply:
Yes, I think you have a point, the only request type we would like to continue is a `REQUEST_TRANSFER`. I'll do some tests and update the PR.